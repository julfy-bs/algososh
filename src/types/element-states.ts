export type ElementStatesVariety = ElementStates.Default | ElementStates.Changing | ElementStates.Modified;

export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}
