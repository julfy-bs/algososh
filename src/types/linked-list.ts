export type LinkedListState =
  LinkedListStateVariety.AddToHead
  | LinkedListStateVariety.AddToTail
  | LinkedListStateVariety.AddByIndex
  | LinkedListStateVariety.DeleteFromHead
  | LinkedListStateVariety.DeleteFromTail
  | LinkedListStateVariety.DeleteByIndex
  | LinkedListStateVariety.Empty
  | LinkedListStateVariety.Success

export enum LinkedListStateVariety {
  AddToHead = 'addToHead',
  AddToTail = 'addToTail',
  AddByIndex = 'addByIndex',
  DeleteFromHead = 'deleteFromHead',
  DeleteFromTail = 'deleteFromTail',
  DeleteByIndex = 'deleteFromIndex',
  Empty = 'null',
  Success = 'success'
}
