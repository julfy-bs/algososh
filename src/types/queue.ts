export type QueueSettings<T> = {
  array: (T | string)[];
  head: number;
  tail: number;
}
