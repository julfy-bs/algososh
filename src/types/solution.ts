export type SolutionState =
  SolutionStateVariety.Add
  | SolutionStateVariety.Delete
  | SolutionStateVariety.Clear
  | SolutionStateVariety.Empty

export enum SolutionStateVariety {
  Add = 'add',
  Delete = 'delete',
  Clear = 'clear',
  Empty = 'null'
}
