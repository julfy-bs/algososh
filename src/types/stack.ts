export type StackSettings = {
  array: string[];
}
