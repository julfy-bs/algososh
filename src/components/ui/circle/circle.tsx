import { FC, ReactElement } from 'react';
import { ElementStates } from '../../../types/element-states';
import styles from './circle.module.css';
import { CircleClassNames } from './utils/circle-class-names';

type CircleProps = {
  state?: ElementStates;
  letter?: string;
  head?: string | ReactElement | null;
  index?: number;
  tail?: string | ReactElement | null;
  tailType?: 'string' | 'element';
  extraClass?: string;
  isSmall?: boolean;
}

export const Circle: FC<CircleProps> = ({
  state = ElementStates.Default,
  letter,
  head,
  index,
  tail,
  extraClass = '',
  isSmall,
}) => {
  return (
    <div className={ `${ styles[CircleClassNames.Content] } ${ extraClass }` }>
      <div
        className={ `
          text
          text_type_input
          text_color_input
          mb-4
          ${ styles.absolute }
          ${ styles[CircleClassNames.Head] } 
          ${ styles[typeof head === 'string' ? 'string' : 'element'] }
        ` }
      >
        { head }
      </div>
      <div
        className={ `${ styles[CircleClassNames.Circle] }  ${ isSmall ? styles.small : '' } ${
          styles[state]
        }` }
      >
        <p
          className={ `text text_type_circle text_color_input ${ styles[CircleClassNames.Letter] }` }
        >
          { letter }
        </p>
      </div>
      <p
        className={ `text text_type_input text_color_input mt-4 ${ styles.absolute } ${ styles[CircleClassNames.Index] }` }
      >
        { index?.toString() }
      </p>
      <div
        className={ `
          text
          text_type_input
          text_color_input
          mt-4 
          ${ styles.absolute }
          ${ index?.toString() ? styles.tail60 : styles.tail30 } 
          ${ styles[typeof tail === 'string' ? 'string' : 'element'] }
        ` }
      >
        { tail }
      </div>
    </div>
  );
};
