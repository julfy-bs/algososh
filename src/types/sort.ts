import { ElementStatesVariety } from './element-states';

export enum Type {
  Bubble = 'bubble',
  Selection = 'selection',
}

export enum Direction {
  Ascending = "ascending",
  Descending = "descending",
}

export type SortType = Type.Bubble | Type.Selection;
export type SortDirection = Direction.Ascending | Direction.Descending;


export type SortElement = {
  readonly id: string;
  value: number;
  state: ElementStatesVariety;
}
