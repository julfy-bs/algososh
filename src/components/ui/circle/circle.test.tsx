import renderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

describe('Компонент <Circle /> рендерится без ошибок', () => {
  it('без буквы', () => {
    const button = renderer
      .create(<Circle />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('с буквами', () => {
    const button = renderer
      .create(<Circle letter={ 'A' } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('с head', () => {
    const button = renderer
      .create(<Circle head={ 'head' } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('с react-элементом в head', () => {
    const button = renderer
      .create(<Circle head={ <Circle isSmall={ true } /> } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('с tail', () => {
    const button = renderer
      .create(<Circle tail={ 'tail' } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('с react-элементом в tail', () => {
    const button = renderer
      .create(<Circle tail={ <Circle isSmall={ true } /> } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('с index', () => {
    const button = renderer
      .create(<Circle index={ 0 } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('с пропсом isSmall ===  true', () => {
    const button = renderer
      .create(<Circle isSmall={ true } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it(`в состоянии ${ ElementStates.Default }`, () => {
    const button = renderer
      .create(<Circle state={ ElementStates.Default } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it(`в состоянии ${ ElementStates.Changing }`, () => {
    const button = renderer
      .create(<Circle state={ ElementStates.Changing } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it(`в состоянии ${ ElementStates.Modified }`, () => {
    const button = renderer
      .create(<Circle state={ ElementStates.Modified } />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
});
