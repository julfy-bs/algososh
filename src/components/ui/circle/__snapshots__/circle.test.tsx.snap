// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <Circle /> рендерится без ошибок без буквы 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок с head 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          string
        "
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок с index 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail60 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  >
    <div
      className="content "
    >
      <div
        className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  >
    <div
      className="content "
    >
      <div
        className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
      />
    </div>
  </div>
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок с tail 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          string
        "
  >
    tail
  </div>
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок с буквами 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;

exports[`Компонент <Circle /> рендерится без ошибок с пропсом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="
          text
          text_type_input
          text_color_input
          mb-4
          absolute
          head 
          element
        "
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="
          text
          text_type_input
          text_color_input
          mt-4 
          absolute
          tail30 
          element
        "
  />
</div>
`;
